### PR TITLE
Improve autofill subdomain/url handling

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203418026431060/f

### Description
Updates the autofill URL matching logic to the linked task. Summarizing:

- better use of e-tld+1 for comparisons
- better handling of subdomains

To achieve this fully, we now filter for matching credentials in two steps:

- at the DB level first using etld+1 
- in Kotlin, where we further filter the results based on subdomains

### Steps to test this PR

Each of the scenarios listed in the spreadsheet should be tested: [Autofill domain matching rules](https://app.asana.com/0/1203115518054842/1203386153863287/f)

- For each row, make sure you have a saved credential in the form listed. Then visit the site detailed for it and validate if autofill is offered, and whether that matches the expected state (✅ meaning we should autofill, ❌ meaning we should not)

### CODE REVIEW

**Comment by CDRussell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2552** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2552" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2552?utm_source=stack-comment).

**Review by karlenDimla — APPROVED** [review]
LGTM! Tested and behaves as indicated in https://app.asana.com/0/608920331025315/1203386153863287/f

File: autofill/autofill-impl/src/test/java/com/duckduckgo/autofill/ui/credential/management/suggestion/SuggestionMatcherTest.kt [hunk 1]
```diff
@@ -69,6 +70,13 @@ class SuggestionMatcherTest {
         assertEquals(2, suggestions.size)
     }
 
+    @Test
+    fun whenSubdomainIncludedInSavedSiteAndVisitingRootSiteThenSuggestionOffered() {
+        val creds = listOf(creds("https://duckduckgo.com"))
+        val suggestions = testee.getSuggestions("https://test.duckduckgo.com", creds)
+        assertEquals(1, suggestions.size)
```
Comment (karlenDimla) [inline autofill/autofill-impl/src/test/java/com/duckduckgo/autofill/ui/credential/management/suggestion/SuggestionMatcherTest.kt:77]: NIT: would be good to compare that the correct  suggestions are being returned than just checking the sise

**Review by karlenDimla — APPROVED** [review]
Looks good!

**Pull request merged**